libnm/nm-device-wifi-p2p.c

```diff
@@ -183,6 +183,151 @@ nm_device_wifi_p2p_get_peer_by_path (NMDeviceWifiP2P *device,
 	return peer;
 }
 
+static void
+start_find_finished_cb (GObject      *obj,
+                        GAsyncResult *res,
+                        gpointer user_data)
+{
+	NMDBusDeviceWifiP2P *proxy = (NMDBusDeviceWifiP2P*) obj;
+	GTask *task = G_TASK (user_data);
+	GError *error = NULL;
+	gboolean success;
+
+	success = nmdbus_device_wifi_p2p_call_start_find_finish (proxy, res, &error);
+	if (!success)
+		g_task_return_error (task, error);
+	else
+		g_task_return_boolean (task, TRUE);
+
+	g_object_unref (task);
+}
+
+/**
+ * nm_device_wifi_p2p_start_find:
+ * @device: a #NMDeviceWifiP2P
+ * @cancellable: a #GCancellable, or %NULL
+ * @callback: a #GAsyncReadyCallback, or %NULL
+ * @user_data: user_data for @callback
+ *
+ * Request NM to search for Wi-Fi P2P peers on @device. Note that the call
+ * returns immediately after requesting the find, and it may take some time
+ * after that for peers to be found.
+ *
+ * The find operation will run for 30s by default. You can stop it earlier
+ * using nm_device_p2p_wifi_stop_find().
+ *
+ * Since: 1.16
+ **/
+void
+nm_device_wifi_p2p_start_find (NMDeviceWifiP2P     *device,
+                               GCancellable        *cancellable,
+                               GAsyncReadyCallback  callback,
+                               gpointer             user_data)
+{
```

Comment:
The D-Bus method also supports an 'options' argument that allows to change the timeout and possibly other things in the future. I think we should add `GVariant *options` argument to the libnm API like we did for `nm_client_add_and_activate_connection2()`.

Reply:
Good point, yes should be done.

Let me merge the branch, as there are no other objections. I'd prefer to keep the time short, where master has an API that we are going to break again (after all, this branch does a significant amount of renaming).